### PR TITLE
Document how to do release

### PR DESCRIPTION
**Pull request by aliok**

### CODE REVIEW

**Comment by aliok** [comment]
ping @matzew @odra @laurafitzgerald @danielpassos 
Guys could you have a short look at this document. It is slightly modified version of https://github.com/aerogear/collateral/wiki/Release-Process-(Java)

I initially was changing that one, but I hated adding comments like "if you're using JBoss Nexus do this, but if you're using Sonatype Nexus, do that". Also, I wanted to make sure the notes are untouched for you guys for things like UPS etc.

File: releasing.md [hunk 1]
```diff
@@ -0,0 +1,304 @@
+# Releasing AeroGear Digger Java Client
+
+### Big Picture
+
+* We release to Sonatype Nexus.
+* That syncs with Maven Central.
+* We don't use JBoss Nexus as it is slower and there is no need for that if we are able to publish to Sonatype.
```
Comment (matzew) [inline releasing.md]: remove that sentence - not need to include this as a potential polemic sentence ...

File: releasing.md [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+    2rS1FJUwRsqi
+    =CoyC
+    -----END PGP PUBLIC KEY BLOCK-----
+```
+
+Copy the block (including the `--------BEGIN...` and `-----END...` lines) and paste it to https://pgp.mit.edu/ and submit.
+
+#### Requirement - Sonatype Nexus account
+
+If you would like to have a fast sync with Maven central, Sonatype's OSS Nexus is the way to go.
```
Comment (matzew) [inline releasing.md]: remove the fast references - no value here

Comment (matzew) [reply releasing.md]: just statte that oss sonatype nexus is used

File: releasing.md [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+
+Set up the password encryption for Maven, you can follow these instructions : http://maven.apache.org/guides/mini/guide-encryption.html
+
+Make sure you created a master password.
+
+### Setup - Sonatype Nexus
+
+First of all, make sure your pom.xml is conforming the requirements of Sonatype: http://central.sonatype.org/pages/apache-maven.html
+
+In order to publish to Sonatype Nexus, have this section in your pom.xml file:
```
Comment (matzew) [inline releasing.md]: _in your_ pom ? this is a doc for `digger-java`, right ?  So I'd guess it's that pom

File: releasing.md [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+mvn release:prepare -Dtag=tag
+```
+
+_**NOTE:**_  The tag should only be the version number like `mvn release:prepare -Dtag=0.2.21`
+
+The release plugin does lots of thing for you like bumping to the next snapshot version and checking files, you can do
+a `git status` and check the differences.
+
+
+#### Sign the Git tag
```
Comment (matzew) [inline releasing.md:209]: I do _like_ signing the tag ++

File: releasing.md [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+```
+
+   * (Optional) Go to your Github profile page and add or make sure you GPG public key is added. Going to [release tab](https://github.com/aerogear/aerogear-parent/releases/tag/0.2.21) you will show a `Verified` label.
+   * If all went OK, push the tag
+```
+git push origin x.y.z
+```
+where x.y.z is the release tag.
+
+**Note:** The delete and recreate tasks are a workaround of this [bug](http://jira.codehaus.org/browse/SCM-486) .
```
Comment (matzew) [inline releasing.md:235]: is that still a thing ?

File: releasing.md [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+
+Now, wait until the staging is done, tests have been successful and community gave approval (or gave no feedback within 72 hours).
+
+
+#### Push the changes performed by maven plugin
+
+**Note:** You only do this step once the staging is done, tests have been successful and community gave approval (or gave no feedback within 72 hours).
+
+```
+git push origin master
```
Comment (matzew) [inline releasing.md:290]: for that you need to disable the auto push setting

File: releasing.md [hunk 7]
```diff
@@ -0,0 +1,304 @@
...
+
+#### Publishing the staged release
+
+**Note:** You only do this step once the staging is done, tests have been successful and community gave approval (or gave no feedback within 72 hours).
+
+The artifact will be literary released. To do that just push the "Release" button next to "Close" button shown above in the screenshots.
+
+Check e.g. here for the release artifacts: https://oss.sonatype.org/content/groups/public/org/jboss/aerogear/digger-java-client/1.1.1/
+
+Sonatype OSS Nexus to synchronize with Maven central repo could take up to 2 days (should be 2-3 hours generally).
```
Comment (matzew) [inline releasing.md]: It's a matter of  minutes :-)

File: releasing.md (hunk 3 above)
Comment (aliok) [reply releasing.md]: yep, right, it is confusing... I started like a more general document and then converted to a Digger Java client specific one :)

File: releasing.md (hunk 5 above)
Comment (aliok) [reply releasing.md:235]: I think so. The tag is not signed w/o that workaround.
That JIRA link is dead... So, cannot check if that's fixed in a future version.
Let's keep it as is: I don't want to do modifications in parent pom(s)

File: releasing.md (hunk 6 above)
Comment (aliok) [reply releasing.md:290]: Well, in my tests, nothing were pushed automatically. So, I assume it is disabled on either AeroGear parent or JBoss parent

**Comment by aliok** [comment]
@matzew pushed some fixes

File: releasing.md (hunk 5 above)
Comment (danielpassos) [reply releasing.md:235]: Yeap, it still needed!

File: releasing.md [hunk 8]
```diff
@@ -0,0 +1,302 @@
...
+    </server>
+    ...
+  </servers>
+</settings>
+
+```
+
+## Testing by releasing snapshots
+
+You can test if things work correctly by releasing a snapshot.
```
Comment (laurafitzgerald) [inline releasing.md]: Not sure of the use of the wore 'things' here. Maybe use, `The release process can be tested by releasing a snapshot`

File: releasing.md [hunk 9]
```diff
@@ -0,0 +1,302 @@
...
+
+This tests two things:
```
Comment (laurafitzgerald) [inline releasing.md]: same here. maybe things -> steps?

**Review by laurafitzgerald — COMMENTED** [review]
Hi @aliok Small in-inline comments. I'm not familiar with the process but this change looks great. I like the addition of the images help with presenting the process. Question for you: Is it possible to automate any of this?

File: releasing.md (hunk 8 above)
Comment (aliok) [reply releasing.md]: makes sense

File: releasing.md (hunk 9 above)
Comment (aliok) [reply releasing.md]: makes sense too :)

**Comment by aliok** [comment]
@laurafitzgerald Releasing process is easy to automate (at least releasing the snapshots), except one must have the credentials on the CI/CD system. So, it might be a bit problematic to use a public service provider like Travis. We can have some job in our private Jenkins, but I don't like it because it is not open :)

Perhaps we should create a JIRA for that and then ask for help from the wider group and the community.

What do you say?

**Comment by aliok** [comment]
@laurafitzgerald @matzew can I get your approvals to merge this?

**Comment by matzew** [comment]
+1

**Pull request merged**